packages/mermaid/src/diagrams/common/common.ts

```diff
@@ -178,13 +178,15 @@ export const getMin = function (...values: number[]): number {
  * @returns The converted string
  */
 export const parseGenericTypes = function (text: string): string {
+  if (!text) {
+    return '';
+  }
+
   let cleanedText = text;
 
   if (text.split('~').length - 1 >= 2) {
     let newCleanedText = cleanedText;
 
-    // use a do...while loop instead of replaceAll to detect recursion
-    // e.g. Array~Array~T~~
```

Comment:
What's the point of removing this comment?

Reply:
I believed it was an old comment that had already been changed..

Reply:
Hmmmmm, it might be worth keeping this comment in.

I could see why somebody would see a [`String.prototype.replace()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace) in a loop, and try to replace it with a [`String.prototype.replaceAll()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll), especially since `.replaceAll()` was only recently added to JavaScript.

Reply:
restored

Reply:
> restored

I think the comment is still missing. Maybe it was re-removed by one of Git's auto-merge commits? It's pretty minor though, so although I'd like to see the comment kept, it's not anything that would block this PR.